locking/locks.go

```diff
@@ -218,100 +219,108 @@ func (c *Client) SearchLocks(filter map[string]string, limit int, localOnly bool
 			return []Lock{}, errors.New("can't search cached locks when filter or limit is set")
 		}
 
-		cacheFile, err := c.prepareCacheDirectory()
-		if err != nil {
-			return []Lock{}, err
-		}
-
-		_, err = os.Stat(cacheFile)
-		if err != nil {
-			if os.IsNotExist(err) {
-				return []Lock{}, errors.New("no cached locks present")
-			}
-
-			return []Lock{}, err
-		}
-
-		return c.readLocksFromCacheFile(cacheFile)
+		locks := []Lock{}
+		err := c.readLocksFromCacheFile("remote", func(decoder *json.Decoder) error {
+			return decoder.Decode(&locks)
+		})
+		return locks, err
 	} else {
 		locks, err := c.searchRemoteLocks(filter, limit)
 		if err != nil {
 			return locks, err
 		}
 
 		if len(filter) == 0 && limit == 0 {
-			cacheFile, err := c.prepareCacheDirectory()
-			if err != nil {
-				return locks, err
-			}
-
-			err = c.writeLocksToCacheFile(cacheFile, locks)
+			err = c.writeLocksToCacheFile("remote", func(writer io.Writer) error {
+				return c.EncodeLocks(locks, writer)
+			})
 		}
 
 		return locks, err
 	}
 }
 
-func (c *Client) VerifiableLocks(ref *git.Ref, limit int) (ourLocks, theirLocks []Lock, err error) {
-	if ref == nil {
-		ref = c.RemoteRef
-	}
-
+func (c *Client) SearchLocksVerifiable(limit int, cached bool) (ourLocks, theirLocks []Lock, err error) {
 	ourLocks = make([]Lock, 0, limit)
 	theirLocks = make([]Lock, 0, limit)
-	body := &lockVerifiableRequest{
-		Ref:   &lockRef{Name: ref.Refspec()},
-		Limit: limit,
-	}
 
-	c.cache.Clear()
+	if cached {
+		if limit != 0 {
+			return []Lock{}, []Lock{}, errors.New("can't search cached locks when limit is set")
+		}
 
-	for {
-		list, res, err := c.client.SearchVerifiable(c.Remote, body)
-		if res != nil {
-			switch res.StatusCode {
-			case http.StatusNotFound, http.StatusNotImplemented:
-				return ourLocks, theirLocks, errors.NewNotImplementedError(err)
-			case http.StatusForbidden:
-				return ourLocks, theirLocks, errors.NewAuthError(err)
-			}
+		locks := &lockVerifiableList{}
+		err := c.readLocksFromCacheFile("verifiable", func(decoder *json.Decoder) error {
+			locks = &lockVerifiableList{}
```

Comment:
Why do we need to set this value to an empty value before decoding?

Reply:
That actually redundant. Fixed. Thanks!